### ISSUE TITLE
训练出来，完全不可用

### ISSUE THREAD

**Issue by tangflash** [netease-youdao/EmotiVoice#72]
训练出来，完全不可用。像哑巴一样，呀呀........，是哈原因差这么远

**Comment by syq163** [comment]
Congratulations on your prompt actions regarding voice cloning. It would be helpful if you could provide more details, such as the data you are using, the number of training steps you have completed, and so on.

**Comment by tangflash** [comment]
1.我按这个步骤中所说的数据进行训练的：https://github.com/netease-youdao/EmotiVoice/tree/main/data/DataBaker。
其中这两项没执行。
![image](https://github.com/netease-youdao/EmotiVoice/assets/5391541/3ba35a5a-15ab-4fc6-a138-d04ffa5cde9c)
2.此外我改为本地模型训练，修改成以下代码：
train_dataset = Dataset_PromptTTS_JETS(config.train_data_path, config, style_encoder)
    #data_sampler = DistributedSampler(train_dataset)
    #train_loader = torch.utils.data.DataLoader(
    #    train_dataset,
    #    num_workers=8,
    #    shuffle=False,
    #    batch_size=config.batch_size,
    #    collate_fn=train_dataset.TextMelCollate,
    #    sampler = data_sampler,
    #)
    # 不再使用 DistributedSampler
    train_loader = torch.utils.data.DataLoader(
        train_dataset,
        #num_workers=8,
        num_workers=1,
        shuffle=True,  # 使用shuffle而不是sampler
        batch_size=config.batch_size,
        collate_fn=train_dataset.TextMelCollate
    )

3.训练了5000步和10000步是一样的效果，没有正常的声音
4.400步val
![val_4000_melspec_0](https://github.com/netease-youdao/EmotiVoice/assets/5391541/0ee3d0b1-9d38-4dc8-ac53-72ea3aa80f57)
![val_4000_melspec_1](https://github.com/netease-youdao/EmotiVoice/assets/5391541/303d4597-28c0-489e-9924-919ee38b802f)

**Comment by syq163** [comment]
> 1.我按这个步骤中所说的数据进行训练的：[https://github.com/netease-youdao/EmotiVoice/tree/main/data/DataBaker。](https://github.com/netease-youdao/EmotiVoice/tree/main/data/DataBaker%E3%80%82) 其中这两项没执行。 ![image](https://private-user-images.githubusercontent.com/5391541/290429463-3ba35a5a-15ab-4fc6-a138-d04ffa5cde9c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI1Mzg3MDcsIm5iZiI6MTcwMjUzODQwNywicGF0aCI6Ii81MzkxNTQxLzI5MDQyOTQ2My0zYmEzNWE1YS0xNWFiLTRmYzYtYTEzOC1kMDRmZmE1Y2RlOWMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMTRUMDcyMDA3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjRhNTA0MjZmOTMzY2Q2MmE5ODBmY2Y3ZjJhODk0ZWM1OTJlMmNmMTBhYjQ4OGY3NWExNTNkM2NjOTc3ZDQ0NSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.NzyRJ4VsVkElPNB5Jr_wgL9-GMC4LPkfvQqbQFWdpU8) 2.此外我改为本地模型训练，修改成以下代码： train_dataset = Dataset_PromptTTS_JETS(config.train_data_path, config, style_encoder) #data_sampler = DistributedSampler(train_dataset) #train_loader = torch.utils.data.DataLoader( # train_dataset, # num_workers=8, # shuffle=False, # batch_size=config.batch_size, # collate_fn=train_dataset.TextMelCollate, # sampler = data_sampler, #) # 不再使用 DistributedSampler train_loader = torch.utils.data.DataLoader( train_dataset, #num_workers=8, num_workers=1, shuffle=True, # 使用shuffle而不是sampler batch_size=config.batch_size, collate_fn=train_dataset.TextMelCollate )
> 
> 3.训练了5000步和10000步是一样的效果，没有正常的声音 4.4000步val ![val_4000_melspec_0](https://private-user-images.githubusercontent.com/5391541/290430263-0ee3d0b1-9d38-4dc8-ac53-72ea3aa80f57.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI1Mzg3MDcsIm5iZiI6MTcwMjUzODQwNywicGF0aCI6Ii81MzkxNTQxLzI5MDQzMDI2My0wZWUzZDBiMS05ZDM4LTRkYzgtYWM1My03MmVhM2FhODBmNTcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMTRUMDcyMDA3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTRiYjU1NDZlM2IxNWY0ODEyNGY4ZjAwODdjNzJlNjkwYWE5MzZiMGU3ODIxYmE5MWQ3M2EzM2EwYWViNTM0OSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.gTmCEenqrkNhk91-UsTJYr9jlprTaYAJ5EMMk_qJSBU) ![val_4000_melspec_1](https://private-user-images.githubusercontent.com/5391541/290430306-303d4597-28c0-489e-9924-919ee38b802f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI1Mzg3MDcsIm5iZiI6MTcwMjUzODQwNywicGF0aCI6Ii81MzkxNTQxLzI5MDQzMDMwNi0zMDNkNDU5Ny0yOGMwLTQ4OWUtOTkyNC05MTllZTM4YjgwMmYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMTRUMDcyMDA3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MWM0MTA2NDQ1NzllNDM2OTkzZjk2YTk4ZDUyYTgxYzk2ZWRhZDU3Yjg3ZTc4MmQ2YzJjMzg1Y2Q5NmVmMThjZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.n0j_VotBWfGW8_xbNeaoaCnJ8cxfqWUwIv8YOIDKC9I)

For DataBaker, it should work fine. I have attached the results from 5000 steps for your reference.
[DataBaker-g_00005000.zip](https://github.com/netease-youdao/EmotiVoice/files/13671707/DataBaker-g_00005000.zip)
And I will attempt to replicate the issue based on the modifications you have made.

**Comment by tangflash** [comment]
谢谢你的及时回复。我想我要重装下python环境试下，你用哪个版本的pytorch和cuda?

**Comment by syq163** [comment]
Some setups for your reference:
cuda 11.8 torch 2.1.1 python 3.10
cuda 11.7 torch 1.13 python 3.8

**Comment by ttPrivacy** [comment]
想请问一下你的硬件配置是？10000步大概训练了多久？我的训练了快10个小时了还没出结果。

**Comment by tangflash** [comment]
3060 12G的显卡，32G内存，10000步大概2个小时。其实5000步就可推理了。



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2023年12月17日(周日) 凌晨5:31
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [netease-youdao/EmotiVoice] 训练出来，完全不可用 (Issue #72)




 
想请问一下你的硬件配置是？10000步大概训练了多久？我的训练了快10个小时了还没出结果。

**Comment by set-path** [comment]
> 训练出来，完全不可用。像哑巴一样，呀呀........，是哈原因差这么远

请问您解决了吗？我遇到了一样的问题，使用https://github.com/netease-youdao/EmotiVoice/tree/main/data/LJspeech提供的步骤，没有执行MFA，微调结果基本没有声音。

**Comment by tangflash** [comment]
使用wsI弄个linux



---原始邮件---
发件人: "Wenbin ***@***.***&gt;
发送时间: 2024年1月13日(周六) 晚上10:29
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [netease-youdao/EmotiVoice] 训练出来，完全不可用 (Issue #72)




  
训练出来，完全不可用。像哑巴一样，呀呀........，是哈原因差这么远
  
请问您解决了吗？我遇到了一样的问题，使用https://github.com/netease-youdao/EmotiVoice/tree/main/data/LJspeech提供的步骤，没有执行MFA，微调结果基本没有声音。
